adcpipeline/pipeline.py

```diff
@@ -72,9 +74,13 @@ def method_list(self) -> List[Callable]:
         """
         return self._method_list
 
-    def __init__(self, df: DataFrame, method_settings: List) -> None:
+    def __init__(self, df: DataFrame, method_settings: List, filename: str = None) -> None:
```

Comment:
Maybe beter to be able to provide a relative path instead of hardcoding the `cache` directory?